app/app/Http/Controllers/SignUpController.php

```diff
@@ -52,6 +53,11 @@ public function createUser(Request $request)
             // Generate Confirmation Link after saving the user.
             $confirmationLink = BaseUser::generateConfirmationLink($newUser);
 
+            if (Input::has('after_signup_redirect')) {
+                $confirmationLink.='?after_signup_redirect=/add-location';
+            }
+            echo($confirmationLink);
```

Comment:
Controllers shouldn't echo anything when they're ready to merge.  Feel free to troubleshoot temporarily with echo statements but that isn't suitable for the master branch.